packages/neuron-ui/src/components/NervosDAO/hooks.ts

```diff
@@ -565,16 +565,19 @@ export const useUpdateDepositEpochList = ({
         depositBlockHashes => {
           const recordKeyIdxMap = new Map<string, number>()
           const batchParams: ['getHeader', string][] = []
-          records.forEach((record, idx) => {
+          let index = 0
+          records.forEach((record) => {
             if (!record.depositOutPoint && record.blockHash) {
               batchParams.push(['getHeader', record.blockHash])
-              recordKeyIdxMap.set(record.outPoint.txHash, idx)
+              recordKeyIdxMap.set(record.outPoint.txHash, index)
```

Comment:
If the elements of `recordKeyIdxMap` comes from 2 arrays and the index grows 1 by 1, it could be an array.

Reply:
It is used to find the index of the record. Because the records key's order is different with batchRequest.

Reply:
Personal feeling is that

```javascript
let index = 0
const recordKeyIdxMap = new Map()
items.forEach(item => {
  if (randomTrue()) {
    recordKeyIdxMap.set(item, index)
    index += 1
  }
}
```

is hard to understand its purpose, but

```javascript
const recordKeyIdx = []
items.forEach(item => {
  if (randomTrue()) {
    recordKeyIdx.push(item)
  }
}
```

is clear

Reply:
It needs a loop to get the item's index when I know the item's value.
But for Map, it can easily get the index by `Map.get`.

Reply:
> It needs a loop to get the item's index when I know the item's value. But for Map, it can easily get the index by `Map.get`.

Does `Array.prototype.indexOf` help?

Reply:
I think `Array.prototype.indexOf` is also using a loop to find the index of `array`. Actually, because the length of array will not be large, I will change it to this for more clarity.